### PR TITLE
NFDIV 1173

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-1173

### Change description ###
image
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/app/controller/PostController.ts [hunk 1]
```diff
@@ -83,7 +85,9 @@ export class PostController<T extends AnyObject> {
   }
 
   protected getEventName(req: AppRequest<T>): string {
-    if (req.session.userCase.applicationType === ApplicationType.SOLE_APPLICATION && req.session.isApplicant2) {
+    if (req.session.userCase.state === State.ConditionalOrderDrafted && !req.session.isApplicant2) {
```
Comment (lara-mpt) [inline src/main/app/controller/PostController.ts:88]: this is only for a sole application as for a joint app2 can update conditional order

**Review by lara-mpt — APPROVED** [review]

File: src/main/app/controller/PostController.ts (hunk 1 above)
Comment (velimd) [reply src/main/app/controller/PostController.ts:88]: yh i was confused on this, tbh i dont know if respondent has any pages to update on this so i left it.

Comment (velimd) [reply src/main/app/controller/PostController.ts:88]: ill confirm with BA

Comment (velimd) [reply src/main/app/controller/PostController.ts:88]: done

File: src/main/app/controller/PostController.ts [hunk 2]
```diff
@@ -85,7 +85,11 @@ export class PostController<T extends AnyObject> {
   }
 
   protected getEventName(req: AppRequest<T>): string {
-    if (req.session.userCase.state === State.ConditionalOrderDrafted && !req.session.isApplicant2) {
+    if (
+      req.session.userCase.state === State.ConditionalOrderDrafted &&
+      (req.session.userCase.applicationType === ApplicationType.JOINT_APPLICATION ||
+        (req.session.userCase.applicationType === ApplicationType.SOLE_APPLICATION && !req.session.isApplicant2))
+    ) {
```
Comment (lara-mpt) [inline src/main/app/controller/PostController.ts:92]: might be easier to do 

```
    if (
      req.session.userCase.state === State.ConditionalOrderDrafted &&
        !(req.session.userCase.applicationType === ApplicationType.SOLE_APPLICATION && req.session.isApplicant2))
```

File: src/main/app/controller/PostController.test.ts [hunk 3]
```diff
@@ -274,7 +274,7 @@ describe('PostController', () => {
     expect(res.redirect).toHaveBeenCalledWith('/next-step-url');
   });
 
-  test('triggers update-conditional-order event if case is in ConditionalOrderDrafted', async () => {
+  test('triggers update-conditional-order event if solo case is in ConditionalOrderDrafted and is applicant1', async () => {
```
Comment (lara-mpt) [inline src/main/app/controller/PostController.test.ts:277]: using 'sole' instead of 'solo' is more consistent but just a preference

**Review by lara-mpt — APPROVED** [review]

File: src/main/app/controller/PostController.ts [hunk 4]
```diff
@@ -83,7 +85,12 @@ export class PostController<T extends AnyObject> {
   }
 
   protected getEventName(req: AppRequest<T>): string {
-    if (req.session.userCase.applicationType === ApplicationType.SOLE_APPLICATION && req.session.isApplicant2) {
+    if (
+      req.session.userCase.state === State.ConditionalOrderDrafted &&
+      !(req.session.userCase.applicationType === ApplicationType.SOLE_APPLICATION && req.session.isApplicant2)
```
Comment (linusnorton) [inline src/main/app/controller/PostController.ts:90]: how come? won't this will apply to both joint applications and app2?

Comment (velimd) [reply src/main/app/controller/PostController.ts:90]: not app2 in sole journey

**Review by linusnorton — APPROVED** [review]

**Pull request merged**